fw/http.c

```diff
@@ -3029,15 +3029,61 @@ tfw_h1_rewrite_purge_to_get(struct sk_buff *head)
 	return -ENOMEM;
 }
 
+/**
+ * Removes n bytes from the beginning of the message, iterating over skb chain
+ * if needed.
+ * @return zero on success and error code on error.
+ *
+ * TODO #1535: to be replaced with future version of ss_skb_chop_head_tail()
+ * over all skbs of the message
+ */
+static int
+tfw_h1_chop_leading_crlf(struct sk_buff **head, unsigned int n)
+{
+	struct sk_buff *skb;
+	while (n) {
+		skb = *head;
+		if (likely(skb->len > n))
+			return ss_skb_chop_head_tail(skb, skb, n, 0);
+		if (WARN_ON(skb->next == skb))
+			return -EBADMSG;
+		n -= skb->len;
+		/* We do not use ss_skb_unlink() here to prevent it
+		 * to remove the last skb in the list and to skip
```

Comment:
Perhaps, a little smoother and clearer:
_We do not use ss_skb_unlink() here to avoid the removal of the last skb in the list and to skip extra actions that are unnecessary here._

Reply:
@keshonok Thank you. Accepted.